design_drafts/joint_interfaces_and_transmissions.md

```diff
@@ -35,49 +35,57 @@ and ties in with the use of the [flexible joint states message](https://github.c
-      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
-      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
-      <hardwareInterface>awesome_interface/FooJointInterface</hardwareInterface>
-      <role></role>
+      <actuator>
+        <!-- control input (u) for the joint, i.e. effort/torque value -->
+        <hardwareInterface>hardware_interface/EffortJointCommandInterface</hardwareInterface>
+      </actuator>
+      <sensor>
+        <!-- output state (x) from the joint, e.g. position -->
+        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
+        <!-- output state (x_dot) from the joint, e.g. velocity -->
+        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
+        <!-- output state (x_dot_dot) from the joint, e.g. acceleration -->
+        <hardwareInterface>hardware_interface/AccelerationJointInterface</hardwareInterface>
+      </sensor>
+    </joint>
+    <joint name="joint2">
+      <actuator>
+        <hardwareInterface>hardware_interface/EffortJointCommandInterface</hardwareInterface>
+      </actuator>
+      <sensor>
+        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
+        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
+        <hardwareInterface>hardware_interface/AccelerationJointInterface</hardwareInterface>
+      </actuator>
+    </joint>
+    <joint>
+      <actuator>
+        <!-- control input (u) for the joint, i.e. velocity value -->
+        <hardwareInterface>hardware_interface/VelocityJointCommandInterface</hardwareInterface>
+      </actuator>
+      <sensor>
+        <!-- custom output state (x) for the joint, i.e. some custom value such as current      
+        <hardwareInterface name="current_joint_interface">hardware_interface/CustomValueInterface</hardwareInterface>
+      </sensor>
     </joint>
-    <actuator name="actuator1">
-      <mechanicalReduction>50</mechanicalReduction>
-    </actuator>
 </transmission>
 ```
-The content of a `hardwareInterface`  XML attribute should match up with strings provided by `ros2_control` in the shape of familiar names:
+
+The content of a `hardwareInterface` XML tag should match up with strings provided by `ros2_control` in the shape of familiar names:
 `hardware_interface::PositionJointInterface, hardware_interface::VelocityJointInterface, hardware_interface::EffortJointInterface`.
-By relying on strings internally in the framework but using them mostly through shared names, a standard nomenclature is kept all the while allowing custom extensions in an easy and transparent way.
-Note the in the example above, since all names are merely strings, `awesome_interface/FooJointInterface` - a custom interface - can be used if it is provided by the `RobotHardware`.
```

Comment:
The "Note the..." sentence was supposed to highlight an important difference, let's keep it.